syft/frameworks/torch/tensors/interpreters/additive_shared.py

```diff
@@ -198,7 +198,7 @@ def generate_shares(secret, n_workers, field, random_type):
         random_shares = [random_type(secret.shape) for _ in range(n_workers - 1)]
 
         for share in random_shares:
-            share.random_(field)
+            share.random_(int(-field / 2), int(field / 2) - 1)
```

Comment:
Q: Should we give an warning if the field values has an odd value?
In that case the random interval would be smaller.

Reply:
I don't think we need a warning since we are working in Finite Field of 2 elements i.e field will always have a value of the form 2^m.

Reply:
In that case it should be fine :)